### PR TITLE
Refactor the JWA backend to utilize common code

### PR DESCRIPTION
**Pull request by kimwnasptd**
first step for #5310 

Will move the backend under `/components/crud-web-apps/jupyter/backend`.

All the backend requests will have the following structure:
```json
{
  "some-field": "some-value",

  "status": 200,
  "success": true,
  "user": "kimwnasptd",
  "log": "this field will be present only when the success is false",
}
```

Need to append a commit for tolerations and affinity.

We (Arrikto) are deploying this jupyter web app on MiniKF and have extensively tested it. So @elikatsis @yanniszark and @StefanoFioravanzo will most probably not have something big to propose in the review.

cc @thesuperzapper

### CODE REVIEW

**Comment by kimwnasptd** [comment]
The last commit for supporting Affinity and Tolerations is there. This is ready for review

File: components/crud-web-apps/jupyter/backend/apps/common/routes/patch.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    # start/stop a notebook
+    if STOP_ATTR in request_body:
+        start_stop_notebook(namespace, notebook, request_body)
+
+    return api.success_response()
+
+
+# helper functions
+def start_stop_notebook(namespace, notebook, request_body):
```
Comment (thesuperzapper) [inline components/crud-web-apps/jupyter/backend/apps/common/routes/patch.py:44]: Why does this need to be a separate function?

The patch function is sufficient?

**Comment by thesuperzapper** [comment]
Should we rename the `/jupyter` prefix on the URL to something more generic?

**Comment by thesuperzapper** [comment]
This looks pretty good as a first pass (although I am still sceptical we need a fully separate code path for ROK).

I assume we will implement the changes from https://github.com/kubeflow/kubeflow/pull/5280 when we make the new `notebook-front-end`, as there are lots of backend changes required for those frontend changes.

File: components/crud-web-apps/jupyter/backend/apps/common/routes/patch.py (hunk 1 above)
Comment (kimwnasptd) [reply components/crud-web-apps/jupyter/backend/apps/common/routes/patch.py:44]: I was thinking that If in the future we want to handle other cases in the PATCH request, then each one of them should be its own function and not have all the logic unfolded in the PATCH handler

Comment (kimwnasptd) [reply components/crud-web-apps/jupyter/backend/apps/common/routes/patch.py:44]: I was thinking that If in the future we want to handle other cases in the PATCH request, then each one of them should be its own function and not have all the logic unfolded in the PATCH handler

**Review by kimwnasptd — COMMENTED** [review]
@thesuperzapper 
> Should we rename the /jupyter prefix on the URL to something more generic?

Was thinking of keeping it as is for now until we replace the the code entirely. Then once we have completely replaced the app we can change the naming both in the routes as well in the manifests naming.

**Comment by kimwnasptd** [comment]
I think we can move on with merging this.
Next step will be to submit the PR for the frontend

**Pull request merged**